Apps/Src/ReadTritium.c

```diff
@@ -1,187 +1,195 @@
+ *  Can result in restarting the motor (for hall sensor errors while less than
+ * MOTOR_RESTART_THRESHOLD) or locking the scheduler and entering a
+ * nonrecoverable fault (all other cases)
  * @param   motor_err Bitmap with motor error codes to check
  */
-static void assertTritiumError(tritium_error_code_t motor_err){   
-	static uint8_t hall_fault_cnt = 0; //trip counter, doesn't ever reset
-	static uint8_t motor_fault_cnt = 0;
-
-	Error_ReadTritium = (error_code_t)motor_err; // Store error codes for inspection info
-	if(motor_err == T_NONE) return; // No error, return
-                                                                                    // NOTE: If we had >1 recoverable errors,
-    // Hall sensor error is the only recoverable error, so any other error          // make sure a combination of them wouldn't 
-    // or combination of errors includes at least one that is nonrecoverable        // accidentally fall into this nonrecoverable bucket
-	if(motor_err != T_HALL_SENSOR_ERR && motor_err != T_MOTOR_WATCHDOG_TRIP){                                             
-		// Assert a nonrecoverable error with no callback function- nonrecoverable will kill the motor and infinite loop
-		throwTaskError(Error_ReadTritium, NULL, OPT_LOCK_SCHED, OPT_NONRECOV);
-		return;
-	}
-
-	// If it's purely a hall sensor error, try to restart the motor a few times and then fail out
-
-	if(motor_err == T_HALL_SENSOR_ERR && ++hall_fault_cnt > RESTART_THRESHOLD){  // Threshold has been exceeded
-		// Assert a nonrecoverable error that will kill the motor, display a fault screen, and infinite loop
-		throwTaskError(Error_ReadTritium, NULL, OPT_LOCK_SCHED, OPT_NONRECOV);
-		return;
-	}
-
-	//try to restart the motor a few times and then fail out
-	if(motor_err == T_MOTOR_WATCHDOG_TRIP && ++motor_fault_cnt > RESTART_THRESHOLD){  
-		// Assert a nonrecoverable error that will kill the motor, display a fault screen, and infinite loop
-		throwTaskError(Error_ReadTritium, NULL, OPT_LOCK_SCHED, OPT_NONRECOV);
-		return;
-	}
-
-	// Threshold hasn't been exceeded, so assert a recoverable error with the motor restart callback function
-	throwTaskError(Error_ReadTritium, handler_ReadTritium_HallError, OPT_NO_LOCK_SCHED, OPT_RECOV); 
-	
-	Error_ReadTritium = T_NONE; // Clear the error after handling it
+static void assertTritiumError(TritiumErrorCode motor_err) {
+    static uint8_t hall_fault_cnt = 0;  // trip counter, doesn't ever reset
+    static uint8_t motor_fault_cnt = 0;
+
+    error_read_tritium =
+        (ErrorCode)motor_err;  // Store error codes for inspection info
+    if (motor_err == kNone) {
+        return;  // No error, return
+    }
+    // NOTE: If we had >1 recoverable errors,
+    // Hall sensor error is the only recoverable error, so any other error //
+    // make sure a combination of them wouldn't or combination of errors
+    // includes at least one that is nonrecoverable        // accidentally fall
+    // into this nonrecoverable bucket
```

Comment:
I wrote this note awhile ago, but I think it no longer applies and can be deleted now
```suggestion
```